### PR TITLE
Update local ruby and add circleci.

### PR DESCRIPTION
**Pull request by philip-maina**
## Summary
This PR updates local ruby (to `3.2.1`) and ruby matrix.

### CODE REVIEW

**Review by prabeshshrestha — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -5,3 +5,10 @@ rvm:
   - 2.5.7
   - 2.6.5
   - 2.7.0
+  - 2.7.5
+  - 3.0.0
```
Comment (sebaherrera07) [inline .travis.yml:9]: I guess with one line for each of the middle version would be OK?
Meaning, you can remove:
```
2.7.0
3.0.0
3.1.0
3.2.0
```

Comment (sebaherrera07) [reply .travis.yml:9]: Also, maybe you can upgrade the bundler version which is way toooo old 😅

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    jobs:
+      - test:
+          matrix:
+            alias: old-ruby
+            parameters:
+              ruby: ["2.4", "2.5", "2.6", "2.7", "3.0", "3.1"]
+          filters:
+            tags:
+              only:
+                - /^v.*/
```
Comment (philip-maina) [inline .circleci/config.yml:64]: Currently, there are no tags (which is surprising being that it is a [published gem](https://rubygems.org/gems/rainforest_auth)). But I'll leave this here in the event we create releases.

**Pull request merged**